src/main/drivers/rangefinder/rangefinder_lidarmt.c

```diff
@@ -105,4 +115,60 @@ const MTRangefinderConfig* getMTRangefinderDeviceConf(rangefinderType_e mtRangef
     return NULL;
 }
 
+const mtRangefinderData_t * getMTRangefinderData(void) {
+    return &rfSensorData;
+}
+
+typedef struct __attribute__((packed)) {
+    uint8_t quality;    // [0;255]
+    int32_t motionX;
+    int32_t motionY;
+} mtOpticalflowDataMessage_t;
+
+static void mtOpticalflowInit(opticalflowDev_t * dev) {
+    UNUSED(dev);
+}
+
+static void mtOpticalflowUpdate(opticalflowDev_t * dev) {
+    UNUSED(dev);
+}
+
+static void mtOpticalflowGetData(opticalflowDev_t * dev, opticalflowData_t * result) {
+    UNUSED(dev);
+    *result = opticalflowSensorData;
+}
+
+bool mtOpticalflowDetect(opticalflowDev_t * dev, rangefinderType_e mtRangefinderToUse) {
+    deviceConf = getMTRangefinderDeviceConf(mtRangefinderToUse);
+    if (!deviceConf) {
+        return false;
+    }
+
+    dev->delayMs = deviceConf->delayMs;
+    dev->minRangeCm = MT_OPTICALFLOW_MIN_RANGE;
+    dev->minQualityThreshold = MT_OPFLOW_MIN_QUALITY_THRESHOLD;
+
+    dev->init   = &mtOpticalflowInit;
+    dev->update = &mtOpticalflowUpdate;
+    dev->read   = &mtOpticalflowGetData;
+    
+    return true;
+}
+
+void mtOpticalflowReceiveNewData(const uint8_t * bufferPtr) {
+    const mtRangefinderData_t * latestRangefinderData = getMTRangefinderData();
+    
+    const mtOpticalflowDataMessage_t * pkt = (const mtOpticalflowDataMessage_t *)bufferPtr;
+    
+    opticalflowSensorData.timeStampUs = micros();
+    opticalflowSensorData.flowRate.x  = (float)pkt->motionX / 1000.0f; 
+    opticalflowSensorData.flowRate.y  = (float)pkt->motionY / 1000.0f;
+    opticalflowSensorData.quality     = (int16_t)pkt->quality * 100 / 255;
```

Comment:
To ensure that I understand what you mean, how would you do this?

Reply:
`pkt->quality * 100 / 255;` or `(singed)pkt->quality * 100 / 255;`, (or maybe `(int)pkt->quality * 100 / 255;` ). When reading code, you assume that explicit cast is important - why is quality converted to `int16_t`?
In this case, the cast is related to `quality` type, so it is fine. But actual C interpretation of the code is different (and result is identical)

Reply:
opticalflowSensorData.quality should be  int16_t to be set negative values in case of sensor failure or out of range

so pkt->quality * 100 / 255; is better ?